File: cine/AMarotoChivite-Cine/AMarotoChivite-Cine-Java/src/main/java/simulacionCine/models/Cliente.java
package simulacionCine.models;

import simulacionCine.enums.EstadoTicket;

// Clase que representa un cliente una vez realice una reserva o compra
public class Cliente {
    private final String nombre;
    private final String apellido;
    private final String telefono;
    public String dni;
    public String email;
    public Ticket informacionTicket;
    private String tarjetaCredito;

    public Cliente(String nombre, String apellido, String dni, String telefono, String email, String tarjetaCredito, Ticket informacionTicket) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.dni = dni;
        this.telefono = telefono;
        this.email = email;
        this.tarjetaCredito = tarjetaCredito;
        this.informacionTicket = informacionTicket;
    }

    /**
     * Devuelve el ID de la sala asociada al ticket del cliente.
     *
     * @return ID de la sala asociada al ticket del cliente.
     */
    public String getIDsalaAsociadaCliente() {
        return informacionTicket.salaID;
    }

    /**
     * Establece la tarjeta de crédito del cliente.
     *
     * @param tarjetaCliente Tarjeta de crédito del cliente.
     */
    public void setTarjetaCredito(String tarjetaCliente) {
        this.tarjetaCredito = tarjetaCliente;
    }

    /**
     * Establece el estado del ticket asociado al cliente.
     *
     * @param estadoTicket Estado del ticket asociado al cliente.
     */
    public void setEstadoTicketAsociado(EstadoTicket estadoTicket) {
        this.informacionTicket.estado = estadoTicket;
    }

    /**
     * Devuelve una cadena con información del cliente.
     *
     * @return Cadena con información del cliente.
     */
    @Override
    public String toString() {
        return "Cliente -> (" +
                "nombre='" + nombre + '\'' +
                ", apellido='" + apellido + '\'' +
                ", telefono='" + telefono + '\'' +
                ", dni='" + dni + '\'' +
                ", email='" + email + '\'' +
                ", tarjetaCredito='" + tarjetaCredito + '\'' +
                "\n" + informacionTicket + '\'' +
                ')';
    }
}

File: cine/AMarotoChivite-Cine/AMarotoChivite-Cine-Kotlin/src/main/kotlin/simulacionCine/models/Cliente.kt
package simulacionCine.models

import simulacionCine.enums.EstadoTicket

// Clase que representa un cliente una vez realice una reserva o compra
class Cliente(
    private var nombre: String,
    private var apellido: String,
    var dni: String,
    private var telefono: String,
    var email: String,
    private var tarjetaCredito: String,
    var informacionTicket: Ticket
) {

    /**
     * Devuelve el ID de la sala asociada al ticket del cliente.
     *
     * @return ID de la sala asociada al ticket del cliente.
     */
    fun getIDsalaAsociadaCliente(): String {
        return informacionTicket.salaID
    }

    /**
     * Establece la tarjeta de crédito del cliente.
     *
     * @param tarjetaCliente Tarjeta de crédito del cliente.
     */
    fun setTarjetaCredito(tarjetaCliente: String) {
        this.tarjetaCredito = tarjetaCliente
    }

    /**
     * Establece el estado del ticket asociado al cliente.
     *
     * @param estadoTicket Estado del ticket asociado al cliente.
     */
    fun setEstadoTicketAsociado(estadoTicket: EstadoTicket) {
        this.informacionTicket.estado = estadoTicket
    }

    /**
     * Devuelve una cadena con información del cliente.
     *
     * @return Cadena con información del cliente.
     */
    override fun toString(): String {
        return "Cliente/Usuario -> (nombre='$nombre', apellido='$apellido', dni='$dni', telefono='$telefono', email='$email', tarjetaCredito='$tarjetaCredito',ticketID='${informacionTicket.ticketIDstring()}')"
    }
}